operator-framework-core/src/test/java/io/javaoperatorsdk/operator/processing/event/source/inbound/CachingInboundEventSourceTest.java

```diff
@@ -0,0 +1,89 @@
+package io.javaoperatorsdk.operator.processing.event.source.inbound;
+
+import java.util.Set;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.javaoperatorsdk.operator.TestUtils;
+import io.javaoperatorsdk.operator.processing.event.EventHandler;
+import io.javaoperatorsdk.operator.processing.event.ResourceID;
+import io.javaoperatorsdk.operator.processing.event.source.AbstractEventSourceTestBase;
+import io.javaoperatorsdk.operator.processing.event.source.CacheKeyMapper;
+import io.javaoperatorsdk.operator.processing.event.source.SampleExternalResource;
+import io.javaoperatorsdk.operator.sample.simple.TestCustomResource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class CachingInboundEventSourceTest extends
+    AbstractEventSourceTestBase<CachingInboundEventSource<SampleExternalResource, TestCustomResource>, EventHandler> {
+
+  public static final int PERIOD = 150;
+  private CachingInboundEventSource.ResourceFetcher<SampleExternalResource, TestCustomResource> supplier =
+      mock(
+          CachingInboundEventSource.ResourceFetcher.class);
+  private TestCustomResource testCustomResource = TestUtils.testCustomResource();
+  private CacheKeyMapper<SampleExternalResource> cacheKeyMapper =
+      r -> r.getName() + "#" + r.getValue();
+
+  @BeforeEach
+  public void setup() {
+    when(supplier.fetchResources(any()))
+        .thenReturn(Set.of(SampleExternalResource.testResource1()));
+
+    setUpSource(new CachingInboundEventSource<>(supplier,
+        SampleExternalResource.class, cacheKeyMapper));
+  }
+
+  @Test
+  void getSecondaryResourceFromCacheOrSupplier() throws InterruptedException {
+    when(supplier.fetchResources(any()))
+        .thenReturn(Set.of(SampleExternalResource.testResource1()));
+
+    var value = source.getSecondaryResources(testCustomResource);
+
+    verify(supplier, times(1)).fetchResources(eq(testCustomResource));
+    verify(eventHandler, never()).handleEvent(any());
+    assertThat(value).hasSize(1);
+
+    value = source.getSecondaryResources(testCustomResource);
+
+    assertThat(value).hasSize(1);
+    verify(supplier, times(1)).fetchResources(eq(testCustomResource));
+    verify(eventHandler, never()).handleEvent(any());
+
+    source.handleResourceEvent(ResourceID.fromResource(testCustomResource),
```

Comment:
@csviri Had to use the Set.of version to avoid replacing the existing secondary. I guess the correct behavior should be additive.

```
source.handleResourceEvent(primaryID, a)
source.handleResourceEvent(primaryID, b)
source.handleResourceEvent(primaryID, c)
source.handleResourceDeleteEvent(primaryID, c)
source.getSecondaryResources(testCustomResource) // should return (a, b)
source.handleResourceEvent(primaryID, Set.of(x,y))
source.getSecondaryResources(testCustomResource) // should return (x,y)
```

Wait for your confirmation and will add javadoc so it's more clear to the user

Reply:
What do you mean by additive exactly?

Reply:
No it should not be. At least that is not what I had on my mind.

Reply:
Probably that would be good to document.
We could add overloaded or rather differerently named versions for additiveness, that would be useful for sure. But this is on purpose, now.